src/main/java/net/minecraftforge/common/tags/ForgeTagBuilder.java

```diff
@@ -173,19 +250,18 @@ public boolean resolve(Function<ResourceLocation, Tag<T>> resourceLocationToTag)
     @Override
     public Tag<T> build(ResourceLocation resourceLocationIn)
     {
-        if (!resolved) LOGGER.trace("Building unresolved Tag!");
-        return factory.build(resourceLocationIn,entries,preserveOrder);
+        return build(resourceLocationIn, comparator);
     }
 
-    public Tag<T> build(ResourceLocation resourceLocationIn, Comparator<T> itemComparator)
+    public Tag<T> build(ResourceLocation resourceLocationIn, @Nullable Comparator<T> itemComparator)
     {
         if (!resolved) LOGGER.trace("Building unresolved Tag!");
-        return factory.build(resourceLocationIn,entries,itemComparator);
+        return factory.build(resourceLocationIn, entries, preserveOrder, itemComparator);
     }
 
-    public interface TagFactory<T> {
-        public Tag<T> build(ResourceLocation location, Set<ITagEntry<T>> entries, boolean preserveOrder);
-
-        public Tag<T> build(ResourceLocation location, Set<ITagEntry<T>> entries, Comparator<T> itemComparator);
+    @FunctionalInterface
+    public interface TagFactory<T>
+    {
+        public Tag<T> build(ResourceLocation location, Set<ITagEntry<T>> entries, boolean preserveOrder, @Nullable Comparator<T> itemComparator);
```

Comment:
try rename `item` to something else so that people don't confuse this with minecraft items.